Services/MovieDataService.swift

```diff
@@ -13,61 +13,87 @@ protocol MovieDataServiceProtocol {
     func getMostPopularMoviesList(completion: @escaping (Result<[Movie], MovieServiceError>)-> Void)
 }
 
-enum MovieServiceError: Error {
+enum MovieServiceError: Error, LocalizedError {
     case failedToGetResponse
     case failedToGetData
-    case failedToDecodeData
-    case errorOccurred
+    case failedToDecode
+    
+    var errorDescription: String? {
+        switch self {
+        case .failedToDecode:
+            return "Failed to decode API response"
+        case .failedToGetResponse:
+            return "Failed to get API response"
+        case .failedToGetData:
+            return "Failed to load data"
+        }
+    }
 }
 
 class MovieDataService: MovieDataServiceProtocol {
+    var playingNowPage = 1
+    var mostPopularPage = 1
+    var isPlayingNowRequestCompleted = true
+    var isMostPopularRequestCompleted = true
+    
     func getPlayingNowMoviesList(completion: @escaping (Result<[Movie], MovieServiceError>)-> Void) {
-        guard let url = URL(string: "https://api.themoviedb.org/3/movie/popular?api_key=b2b14caf40262a9c19a366b15e4e3537&language=en-US&page=1") else { return }
-
+        guard isPlayingNowRequestCompleted else { return }
+        guard let url = URL(string: "https://api.themoviedb.org/3/movie/popular?api_key=b2b14caf40262a9c19a366b15e4e3537&language=en-US&page=\(playingNowPage)") else { return }
+        
+        isPlayingNowRequestCompleted = false
         let request = URLRequest(url: url)
         URLSession.shared.dataTask(with: request) {(data, response, error) in
             if let error = error {
                 print(error.localizedDescription)
-//                How can I use error.localizedDescription to display the error in the VC?
-                completion(.failure(.errorOccurred))
+                completion(.failure(.failedToGetResponse))
                 return
             }
-            
+                   
             guard let data = data else {
                 completion(.failure(.failedToGetData))
                 return
             }
+             
+            self.isPlayingNowRequestCompleted = true
```

Comment:
Let's move this to be called as soon as the `dataTask` finishes, on line 46. In the scenario here, in case you get error it will return without setting `isPlayingNowRequestCompleted` back to true making it impossible to fetch more data afterwards.